### PR TITLE
Add Domestic abuse flux chart

### PR DESCRIPTION
**Pull request by AlokDatta**
### Change description ###
Adding flux chart for domestic abuse module

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: apps/da/da-cos-api/da-cos-api.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: da-cos-api
+spec:
+  releaseName: da-cos-api
+  interval: 1m
+  values:
+    java:
+      image: hmctspublic.azurecr.io/da/cos-api:latest* #{"$imagepolicy": "flux-system:adoption-cos-api"}
```
Comment (timja) [inline apps/da/da-cos-api/da-cos-api.yaml:10]: image policy looks wrong? (adoption)

File: apps/da/identity/aat.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: "aadpodidentity.k8s.io/v1"
+kind: AzureIdentity
+metadata:
+  name: da
+  namespace: da
+spec:
+  resourceID: "/subscriptions/1c4f0704-a29e-403d-b719-b90c34ef14c9/resourcegroups/managed-identities-aat-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/prl-aat-mi"
```
Comment (timja) [inline apps/da/identity/aat.yaml:7]: ptl is probably not the right identity?

File: k8s/aat/common-overlay/da/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    kind: RoleBinding
+    name: nonprod-team-permissions
+    namespace: da
+  patch: |-
+    - op: add
+      path: "/subjects"
+      value:
+        - apiGroup: rbac.authorization.k8s.io
+          kind: Group
+          name: "882d8512-688b-48b1-8ce2-05fd49d90478"
```
Comment (timja) [inline k8s/aat/common-overlay/da/kustomization.yaml:20]: is family private law the right group for DA?

File: k8s/aat/common/da/da-cos-api.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      replicas: 2
+      memoryRequests: "768Mi"
+      cpuRequests: "1000m"
+      memoryLimits: "3072Mi"
+      cpuLimits: "2500m"
+      readinessDelay: 45
+      readinessTimeout: 5
+      readinessPeriod: 15
+      useInterpodAntiAffinity: true
+      image: hmctspublic.azurecr.io/da/cos-api:pr-*
```
Comment (timja) [inline k8s/aat/common/da/da-cos-api.yaml:26]: invalid image, use latest if you don't know what to put but it should have an image policy next to it

File: k8s/aat/common/da/da-cos-api.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+apiVersion: helm.fluxcd.io/v1
```
Comment (timja) [inline k8s/aat/common/da/da-cos-api.yaml:1]: this must be kustomized, please follow https://github.com/hmcts/cnp-flux-config/blob/master/docs/app-deployment.md

File: k8s/aat/common/da/identity.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+apiVersion: "aadpodidentity.k8s.io/v1"
+kind: AzureIdentity
+metadata:
+  name: da
+  namespace: da
+spec:
+  type: 0
+  resourceID: "/subscriptions/1c4f0704-a29e-403d-b719-b90c34ef14c9/resourcegroups/managed-identities-aat-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/prl-aat-mi"
```
Comment (timja) [inline k8s/aat/common/da/identity.yaml:8]: prl?

File: k8s/demo/common-overlay/da/kustomization.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    kind: RoleBinding
+    name: nonprod-team-permissions
+    namespace: da
+  patch: |-
+    - op: add
+      path: "/subjects"
+      value:
+        - apiGroup: rbac.authorization.k8s.io
+          kind: Group
+          name: "882d8512-688b-48b1-8ce2-05fd49d90478"
```
Comment (timja) [inline k8s/demo/common-overlay/da/kustomization.yaml:20]: family private law?

File: k8s/namespaces/da/da-cos-api/da-cos-api.yaml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    java:
+      memoryRequests: "768Mi"
+      cpuRequests: "1000m"
+      memoryLimits: "3072Mi"
+      cpuLimits: "2500m"
+      readinessDelay: 45
+      readinessTimeout: 5
+      readinessPeriod: 15
+      useInterpodAntiAffinity: true
+      image: hmctspublic.azurecr.io/da/cos-api:latest #{"$imagepolicy": "flux-system:adoption-cos-api"}
```
Comment (timja) [inline k8s/namespaces/da/da-cos-api/da-cos-api.yaml:25]: adoption?

File: k8s/namespaces/da/da-cos-api/da-cos-api.yaml [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+      # ingressHost: adoption-cos-api.aat.platform.hmcts.net
```
Comment (timja) [inline k8s/namespaces/da/da-cos-api/da-cos-api.yaml:26]: ?

File: k8s/perftest/common/da/identity.yaml [hunk 10]
```diff
@@ -0,0 +1,20 @@
+apiVersion: "aadpodidentity.k8s.io/v1"
+kind: AzureIdentity
+metadata:
+  name: da
+  namespace: da
+spec:
+  type: 0
+  resourceID: "/subscriptions/1c4f0704-a29e-403d-b719-b90c34ef14c9/resourcegroups/managed-identities-aat-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/prl-aat-mi"
```
Comment (timja) [inline k8s/perftest/common/da/identity.yaml:8]: prl?

File: k8s/preview/common-overlay/da/kustomization.yaml [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    kind: RoleBinding
+    name: nonprod-team-permissions
+    namespace: da
+  patch: |-
+    - op: add
+      path: "/subjects"
+      value:
+        - apiGroup: rbac.authorization.k8s.io
+          kind: Group
+          name: "882d8512-688b-48b1-8ce2-05fd49d90478"
```
Comment (timja) [inline k8s/preview/common-overlay/da/kustomization.yaml:21]: family private law?

**Review by timja — COMMENTED** [review]
please take a look at my comments

File: apps/da/da-cos-api/da-cos-api.yaml (hunk 1 above)
Comment (AlokDatta) [reply apps/da/da-cos-api/da-cos-api.yaml]: updated to - image: hmctspublic.azurecr.io/da/cos-api:latest  #{"$imagepolicy": "flux-system:da-cos-api"}

File: apps/da/identity/aat.yaml (hunk 2 above)
Comment (AlokDatta) [reply apps/da/identity/aat.yaml]: Modified to da

File: k8s/aat/common-overlay/da/kustomization.yaml (hunk 3 above)
Comment (AlokDatta) [reply k8s/aat/common-overlay/da/kustomization.yaml]: Modified to Domestic Abuse

File: k8s/aat/common/da/da-cos-api.yaml (hunk 4 above)
Comment (AlokDatta) [reply k8s/aat/common/da/da-cos-api.yaml]: updated to latest with image policy

File: k8s/aat/common/da/identity.yaml (hunk 6 above)
Comment (AlokDatta) [reply k8s/aat/common/da/identity.yaml]: updated to DA

File: k8s/demo/common-overlay/da/kustomization.yaml (hunk 7 above)
Comment (AlokDatta) [reply k8s/demo/common-overlay/da/kustomization.yaml]: updated with DA

File: k8s/namespaces/da/da-cos-api/da-cos-api.yaml (hunk 8 above)
Comment (AlokDatta) [reply k8s/namespaces/da/da-cos-api/da-cos-api.yaml]: updated with DA

File: k8s/namespaces/da/da-cos-api/da-cos-api.yaml (hunk 9 above)
Comment (AlokDatta) [reply k8s/namespaces/da/da-cos-api/da-cos-api.yaml]: Modified to da

File: k8s/perftest/common/da/identity.yaml (hunk 10 above)
Comment (AlokDatta) [reply k8s/perftest/common/da/identity.yaml]: Modified to da

File: k8s/preview/common-overlay/da/kustomization.yaml (hunk 11 above)
Comment (AlokDatta) [reply k8s/preview/common-overlay/da/kustomization.yaml]: Modified to Domestic Abuse

File: k8s/aat/common/da/da-cos-api.yaml (hunk 5 above)
Comment (AlokDatta) [reply k8s/aat/common/da/da-cos-api.yaml:1]: Customisation done as per fluxV2 steps inside apps folder.

Comment (timja) [reply k8s/aat/common/da/da-cos-api.yaml:1]: still not done

**Review by timja — CHANGES_REQUESTED** [review]

**Pull request closed without merging**